demo/demo.py

```diff
@@ -12,12 +12,24 @@
 OBJECTRON_CLASSES = ('bike', 'book', 'bottle', 'cereal_box', 'camera', 'chair', 'cup', 'laptop', 'shoe')
 
 class Detector:
-    """Wrapper class for object detector"""
+    """Wrapper class for face detector"""
     def __init__(self, ie,  model_path, conf=.6, device='CPU', ext_path=''):
         self.net = load_ie_model(ie, model_path, device, None, ext_path)
         self.confidence = conf
         self.expand_ratio = (1., 1.)
 
+    def run_async(self, frame):
+        self.frame_shape = frame.shape
+        self.net.forward_async(frame)
+
+    def wait_and_grab(self):
+        outputs = self.net.grab_all_async()
+        detections = []
+        for out in outputs:
```

Comment:
In this wrapper we suppose we have only one active infer request, se you can just add a check that length of `outputs` is one and don't run this loop at all.